[2025-05-27T19:28:15.512+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: walmart_etl_dag.generate_synthetic_data manual__2025-05-27T19:28:14+00:00 [queued]>
[2025-05-27T19:28:15.517+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: walmart_etl_dag.generate_synthetic_data manual__2025-05-27T19:28:14+00:00 [queued]>
[2025-05-27T19:28:15.517+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 2
[2025-05-27T19:28:15.523+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): generate_synthetic_data> on 2025-05-27 19:28:14+00:00
[2025-05-27T19:28:15.527+0000] {standard_task_runner.py:57} INFO - Started process 1345 to run task
[2025-05-27T19:28:15.534+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'walmart_etl_dag', 'generate_synthetic_data', 'manual__2025-05-27T19:28:14+00:00', '--job-id', '308', '--raw', '--subdir', 'DAGS_FOLDER/walmart_etl_dag.py', '--cfg-path', '/tmp/tmp_j_25k58']
[2025-05-27T19:28:15.539+0000] {standard_task_runner.py:85} INFO - Job 308: Subtask generate_synthetic_data
[2025-05-27T19:28:15.578+0000] {task_command.py:416} INFO - Running <TaskInstance: walmart_etl_dag.generate_synthetic_data manual__2025-05-27T19:28:14+00:00 [running]> on host d80f98a6628f
[2025-05-27T19:28:15.894+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='walmart_etl_dag' AIRFLOW_CTX_TASK_ID='generate_synthetic_data' AIRFLOW_CTX_EXECUTION_DATE='2025-05-27T19:28:14+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-27T19:28:14+00:00'
[2025-05-27T19:28:15.895+0000] {logging_mixin.py:154} INFO - Starting synthetic data generation process...
[2025-05-27T19:28:15.895+0000] {logging_mixin.py:154} INFO - Generating supplier data...
[2025-05-27T19:28:15.916+0000] {logging_mixin.py:154} INFO - Loaded 50 supplier records to ods_supplier
[2025-05-27T19:28:15.916+0000] {logging_mixin.py:154} INFO - Generating return reason data...
[2025-05-27T19:28:15.919+0000] {logging_mixin.py:154} INFO - Loaded 10 return reason records to ods_return_reason
[2025-05-27T19:28:15.919+0000] {logging_mixin.py:154} INFO - Retrieving existing data...
[2025-05-27T19:28:15.921+0000] {logging_mixin.py:154} INFO - Generating inventory data...
[2025-05-27T19:28:15.922+0000] {logging_mixin.py:154} INFO - No products or stores found. Cannot generate inventory data.
[2025-05-27T19:28:15.922+0000] {logging_mixin.py:154} INFO - Generating returns data...
[2025-05-27T19:28:15.923+0000] {logging_mixin.py:154} INFO - Missing required data for returns generation.
[2025-05-27T19:28:15.923+0000] {logging_mixin.py:154} INFO - Transforming and loading data to staging tables...
[2025-05-27T19:28:15.926+0000] {logging_mixin.py:154} INFO - Supplier and return reason data successfully loaded to staging tables with batch ID: BATCH_SYNTHETIC_20250527192815
[2025-05-27T19:28:15.926+0000] {logging_mixin.py:154} INFO - Starting load to target layer for batch BATCH_SYNTHETIC_20250527192815...
[2025-05-27T19:28:15.932+0000] {logging_mixin.py:154} INFO - Supplier and return reason data successfully loaded to target tables
[2025-05-27T19:28:15.932+0000] {logging_mixin.py:154} INFO - Synthetic data generation and loading complete!
[2025-05-27T19:28:15.932+0000] {python.py:194} INFO - Done. Returned value was: None
[2025-05-27T19:28:15.937+0000] {taskinstance.py:1400} INFO - Marking task as SUCCESS. dag_id=walmart_etl_dag, task_id=generate_synthetic_data, execution_date=20250527T192814, start_date=20250527T192815, end_date=20250527T192815
[2025-05-27T19:28:15.967+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2025-05-27T19:28:15.984+0000] {taskinstance.py:2778} INFO - 1 downstream tasks scheduled from follow-on schedule check
